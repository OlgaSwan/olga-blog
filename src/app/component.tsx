import React, { FunctionComponent, useEffect, useState } from 'react'

export const Component: FunctionComponent = () => {
  const [state, setState] = useState('testing')
  const [data, setData] = useState<string | null>(null)
  useEffect(() => {
    setTimeout(() => setData('something'), 100)
  })
  return (
    <>
      <header title='heading'>Component Heading</header>
      <main>
        {data && (
          <div title='data' style={{ border: '1px' }}>
            data
          </div>
        )}
        <input placeholder='Enter text' type='text' value={state} onChange={e => setState(e.target.value)} />
        <button onClick={() => setState('')}>Reset</button>
      </main>
      <footer title='footering'>
        <button placeholder='Disabled' disabled={true}>
          Disabled
        </button>
      </footer>
    </>
  )
}

type ArrElement = number | string | null | undefined
export const mapArrToStrings = (arr: ArrElement[]) => {
  return arr.filter(e => Number.isInteger(e)).map(String)
}

export const square = (number: number) => {
  if (number === 1) return number
  return Math.pow(number, number)
}

type Callback = (a: number, b: number) => number

export const delayFunc: Callback = (x, y) => x + y

export const delay = (callback: Callback, ms: number) =>
  new Promise(resolve => setTimeout(() => resolve(callback(5, 5)), ms))
