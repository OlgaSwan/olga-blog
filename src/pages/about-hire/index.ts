import { createLoadablePage } from '../../shared/loadable-page'

export const AboutHire = createLoadablePage(() => import('./page'))
