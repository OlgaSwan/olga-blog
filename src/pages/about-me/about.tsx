import React, { useContext, useRef } from 'react'
import { gsap } from 'gsap'
import { useGSAP } from '@gsap/react'
import { Box, Image, Paragraph, ResponsiveContext } from 'grommet'

import { handleMouseMove } from 'src/shared/utils/tilting'
import selfie from 'src/pages/about-me/selfie.jpg'

export const About = () => {
  const imageRef = useRef<HTMLDivElement>(null)
  const { contextSafe } = useGSAP({ scope: imageRef })
  const screenSize = useContext(ResponsiveContext)

  const moveCard = contextSafe((rotateX: number, rotateY: number) => {
    gsap.to(imageRef.current, {
      rotationY: rotateY,
      rotationX: rotateX,
      duration: 0.8,
      ease: 'circ.out',
    })
  })

  const resetCard = contextSafe(() => {
    gsap.to(imageRef.current, {
      rotationY: 0,
      rotationX: 0,
      duration: 0.8,
      ease: 'circ.out',
    })
  })

  return (
    <Box direction={screenSize === 'small' ? 'column' : 'row'} gap='small' align='center'>
      <Box gap='medium' width='large'>
        <Paragraph>{`Beyond the technologies, I'm interested in many aspects of life. Sometimes, when I have free time from reading tech documentation, I spend hours with biographical and World War literature.`}</Paragraph>
        <Paragraph>{`Art holds a special place in my heart, helps expressing my creativity through both traditional and digital drawing. The canvas serves for me as a means of conveying emotions and ideas into visual expressions.`}</Paragraph>
        <Paragraph>{`Equally enamored with the simple joys of nature, I'm taking walks outside. In these moments I find clarity, inspiration, and a connection to the world around me.`}</Paragraph>
        <Paragraph>{`I'm an individual who lives on the intersection of technology, empathy and art.`}</Paragraph>
      </Box>
      <Box
        ref={imageRef}
        onMouseMove={e => handleMouseMove(e, imageRef, moveCard)}
        onMouseLeave={() => resetCard()}
        height='420px'
        width='medium'
      >
        <Image fit='cover' src={selfie} style={{ borderRadius: '10px', cursor: 'pointer' }} />
      </Box>
    </Box>
  )
}
