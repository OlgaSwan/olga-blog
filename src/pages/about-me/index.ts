import { createLoadablePage } from '../../shared/loadable-page'

export const AboutMe = createLoadablePage(() => import('./page'))
