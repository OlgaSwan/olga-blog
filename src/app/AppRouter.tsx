import React, { FunctionComponent, useState, useEffect } from "react";
import { BrowserRouter, Routes, Route } from "react-router-dom";

import { AuthContext } from "../model/context/authContext";

import { Home } from "../pages/home";
import { Diary } from "../pages/diary";
import { Login } from "../pages/login";
import { CreateAcc } from "../pages/createaccount";
import { About } from "../pages/about";
import { Uses } from "../pages/uses";

export const AppRouter: FunctionComponent = () => {
  const [isAuth, setIsAuth] = useState(false);

  useEffect(() => {
    if (localStorage.getItem("isAuth") === "true") {
      setIsAuth(true);
    }
  }, []);

  return (
    <AuthContext.Provider
      value={{
        isAuth,
        setIsAuth,
      }}
    >
      <BrowserRouter>
        <Routes>
          <Route path="/" Component={Home} />
          <Route path="/diary/:id" Component={Diary} />
          <Route path="/login" Component={Login} />
          <Route path="/createaccount" Component={CreateAcc} />
          <Route path="/about" Component={About} />
          <Route path="/uses" Component={Uses} />
        </Routes>
      </BrowserRouter>
    </AuthContext.Provider>
  );
};
