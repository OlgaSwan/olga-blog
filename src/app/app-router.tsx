import React, { FunctionComponent } from 'react'
import { createBrowserRouter, Navigate, RouterProvider } from 'react-router-dom'

import { routeMap } from 'src/shared/route-map'

import { Home } from 'src/pages/home'
import { AboutMe } from 'src/pages/about-me'
import { AboutHire } from 'src/pages/about-hire'
import { BlogHome } from 'src/pages/blog-home'
import { BlogDiaryId } from 'src/pages/blog-diary-id'
import { AuthLogin } from 'src/pages/auth-login'
import { AdminHome } from 'src/pages/admin-home'
import { AdminDiaryList } from 'src/pages/admin-diary-list'
import { AdminDiaryId } from 'src/pages/admin-diary-id'
import { AdminTagList } from 'src/pages/admin-tag-list'
import { ErrorForbidden } from 'src/pages/error-forbidden'
import { ErrorNotFound } from 'src/pages/error-not-found'

const router = createBrowserRouter([
  { path: routeMap.home, Component: Home },
  { path: routeMap.aboutMe, Component: AboutMe },
  { path: routeMap.aboutHire, Component: AboutHire },
  { path: routeMap.blogHome, Component: BlogHome },
  { path: routeMap.blogDiaryId(':id'), Component: BlogDiaryId },
  { path: routeMap.authLogin, Component: AuthLogin },
  { path: routeMap.adminHome, Component: AdminHome },
  { path: routeMap.adminDiaryList, Component: AdminDiaryList },
  { path: routeMap.adminDiaryId(':id'), Component: AdminDiaryId },
  { path: routeMap.adminTagList, Component: AdminTagList },
  { path: routeMap.errorForbidden, Component: ErrorForbidden },
  { path: routeMap.errorNotFound, Component: ErrorNotFound },
  { path: '*', element: <Navigate to='/404' /> }
])

export const AppRouter: FunctionComponent = () => <RouterProvider router={router} />
