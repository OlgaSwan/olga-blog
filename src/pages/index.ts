import { Home } from './home'
import { AboutHire } from './about-hire'
import { AboutMe } from './about-me'
import { BlogHome } from './blog-home'
import { BlogDiaryId } from './blog-diary-id'
import { AuthLogin } from './auth-login'
import { AdminHome } from './admin-home'
import { AdminDiaryList } from './admin-diary-list'
import { AdminDiaryId } from './admin-diary-id'
import { AdminTagList } from './admin-tag-list'
import { ErrorNotFound } from './error-not-found'
import { ErrorForbidden } from './error-forbidden'

export const routeMap = {
  home: {
    path: '/',
    Component: Home,
  },
  aboutHire: {
    path: '/about/hire',
    Component: AboutHire,
  },
  aboutMe: {
    path: '/about/me',
    Component: AboutMe,
  },
  blogHome: {
    path: '/blog',
    Component: BlogHome,
  },
  blogHomeTag: {
    path: '/blog/:tag?',
    Component: BlogHome,
  },
  blogDiaryId: {
    path: '/blog/diary/:id',
    Component: BlogDiaryId,
  },
  authLogin: {
    path: '/auth/login',
    Component: AuthLogin,
  },
  adminHome: {
    path: '/admin',
    Component: AdminHome,
  },
  adminDiaryList: {
    path: '/admin/diary',
    Component: AdminDiaryList,
  },
  adminDiaryId: {
    path: '/admin/diary/:id',
    Component: AdminDiaryId,
  },
  adminTagList: {
    path: '/admin/tag',
    Component: AdminTagList,
  },
  errorForbidden: {
    path: '/403',
    Component: ErrorForbidden,
  },
  errorNotFound: {
    path: '/404',
    Component: ErrorNotFound,
  },
}
