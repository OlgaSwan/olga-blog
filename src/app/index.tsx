import React from 'react'
import ReactDOM from 'react-dom/client'
import { HelmetProvider } from 'react-helmet-async'
import 'modern-css-reset'

import { AppRouter } from './app-router'

const root = ReactDOM.createRoot(document.getElementById('root') as HTMLElement)

root.render(
  <React.StrictMode>
    <HelmetProvider>
      <AppRouter />
    </HelmetProvider>
  </React.StrictMode>,
)
