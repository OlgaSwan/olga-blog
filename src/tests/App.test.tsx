import { expect, test, describe } from 'vitest'
import { render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import { Component, delay, delayFunc, mapArrToStrings, square } from '../app/component'
import { AboutMe } from '../pages/about-me'
// Component
describe('Testing component', () => {
  test('heading testing', () => {
    render(<Component />)

    expect(screen.getByTitle('heading')).toBeInTheDocument()
    expect(screen.getByTitle('heading')).toHaveTextContent('Component Heading')
  })
  test('main testing', async () => {
    render(<Component />)

    const divData = await screen.findByTitle('data')
    expect(divData).toBeInTheDocument()
    expect(divData).toHaveStyle({ backgroundColor: 'beige' })

    const resetBtn = screen.getByText('Reset')
    const input = screen.getByPlaceholderText('Enter text')
    await userEvent.type(input, ' example')
    expect(input).toHaveValue('testing example')
    expect(resetBtn).toBeInTheDocument()
    expect(resetBtn).not.toBeDisabled()
    await userEvent.click(resetBtn)
    expect(input).not.toHaveValue(undefined)
    expect(input).toHaveValue('')

    expect(resetBtn).toMatchSnapshot()
    expect(input).toMatchSnapshot()
  })
  test('footer testing', () => {
    render(<Component />)

    expect(screen.getByTitle('footering')).toBeInTheDocument()
    expect(screen.getByPlaceholderText('Disabled')).toBeDisabled()
  })
})

// mapArrToString
describe('Mapping to string', () => {
  test('Number array', () => {
    expect(mapArrToStrings([1, 2, 3])).toStrictEqual(['1', '2', '3'])
  })
  test('Mix array', () => {
    expect(mapArrToStrings([1, 2, 3, 'str', null, undefined])).toStrictEqual(['1', '2', '3'])
  })
})

// describe('square', () => {
//   let mockValue: number
//   beforeAll(() => {
//     mockValue = Math.round(Math.random())
//   })
//   test('Simple', () => {
//     expect(square(mockValue)).toBe(Math.pow(mockValue, mockValue))
//     expect(square(mockValue)).not.toBeUndefined()

//     const spyMathPow = jest.spyOn(Math, 'pow')
//     square(1)
//     expect(spyMathPow).not.toHaveBeenCalled()
//   })

//   afterAll(() => jest.clearAllMocks())
// })

describe('delay', () => {
  test('Delay Function', async () => {
    const sum = await delay(delayFunc, 1000)
    expect(sum).toBe(10)
  })
})

//I'M SO FUCKING DONE eto kakoy-to pizdec
//me too
// describe('Testing router', () => {
//   test('Testing avatar link', async () => {
//     render(<AboutMe />)
//     const link = screen.getByTestId('avatar-link')
//     await userEvent.click(link)
//     expect(screen.getByTestId('home-page')).toBeInTheDocument()
//   })
// })
