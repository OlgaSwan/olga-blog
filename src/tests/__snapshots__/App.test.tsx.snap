// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Testing component > main testing 1`] = `
<button>
  Reset
</button>
`;

exports[`Testing component > main testing 2`] = `
<input
  placeholder="Enter text"
  type="text"
  value=""
/>
`;
